### ISSUE TITLE
[changelog] Changelog is generated even if an external file is specified

### ISSUE THREAD

**Issue by charphi** [jreleaser/jreleaser#965]
### Steps to Reproduce

1. Set an external changelog file in a GitHub release section

### Expected Behaviour

No changelog should be generated.
JReleaser should check that the external file exists during dry-run.

### Actual Behaviour

Changelog is generated.
JReleaser doesn't check that the external file exists during dry-run.

**Comment by aalmiray** [comment]
I can't replicate this error with JReleaser `1.2.0`.

1. git clone https://github.com/aalmiray/foobar.git
2. edit `jreleaser.yml` so that the release section looks like this
```yml
release:
  github:
    overwrite: true
    sign: true
    discussionCategoryName: Announcements
    changelog:
      external: CHANGELOG.md
      preset: conventional-commits
      formatted: ALWAYS
      format: '- {{commitShortHash}} {{commitTitle}}'
      contributors:
        format: '- {{contributorName}}{{#contributorUsernameAsLink}} ({{.}}){{/contributorUsernameAsLink}}'
```
3. Create a file named `CHANGELOG.md` at the root with
```md
## Changelog

 - Fixed stuff
```
4. Run `jreleaser changelog -g`
```
[INFO]  Project version set to 1.0.0
[INFO]  Release is not snapshot
[INFO]  Timestamp is 2022-09-24T14:57:11.769714-05:00
[INFO]  HEAD is at 4df324c
[INFO]  Platform is osx-x86_64
[INFO]  dry-run set to true
[INFO]  Reading changelog from CHANGELOG.md
[INFO]  Generating changelog: out/jreleaser/release/CHANGELOG.md
[DEBUG] ## Changelog

 - Fixed stuff


[INFO]  Calculating checksums
```
6. Run `jreleaser release --dry-run`.

The changelog at `out/jreleaser/release/CHANGELOG.md` matches the content of `CHANGELOG.md` at the root.

**Comment by charphi** [comment]
In my config, I use the `jreleaser-maven-plugin` in a module of a multi-module Maven project.

The external changelog is located on parent/root project and is specified like that:
```xml
<external>${project.basedir}/../CHANGELOG.md</external>
```

**Comment by aalmiray** [comment]
Aha, then I guess what could happen is that the value of `<external>` is resolved to a blank string but that would be really hard given the explicit value you've shown. The maven plugin uses the same model classes as core (in `v1.2.0) thus testing on CLI should yield the same results as testing with the maven plugin. Something else may be at play here.

**Comment by charphi** [comment]
After some digging, I think that the changelog is generated by the [ChangelogWorkflowItem](https://github.com/jreleaser/jreleaser/blob/7b20c673d1afc9c4f40cef6021e860e471d8a7bb/core/jreleaser-engine/src/main/java/org/jreleaser/workflow/ChangelogWorkflowItem.java#L35) whatever the content of JReleaserContext.

**Comment by aalmiray** [comment]
I'm unable to reproduce the problem with `1.2.0`.

**Comment by charphi** [comment]
Strange.
Here is a way to reproduce the problem from a real project:
1. clone the project https://github.com/nbbrd/picocsv 
2. set the env var `JRELEASER_GITHUB_TOKEN` to a dummy value such as `123`
3. run the command `mvn clean install -DskipTests -Pfull-release -Djreleaser.dry.run`
4. the generated changelog `target\jreleaser\release\CHANGELOG.md` is different from the expected changelog `target\CHANGELOG.md`

_Note that the GitHub release page has the expected changelog._

**Comment by aalmiray** [comment]
Found the issue. Thank you for the reproducer! 😅

**Comment by aalmiray** [comment]
Released in v1.3.0 -> https://github.com/jreleaser/jreleaser/releases/tag/v1.3.0
